### PR TITLE
Record the lazy tracing time(C++) in metrics

### PR DESCRIPTION
**Pull request by JackCaoG**
This way we at least know how long it takes to trace the op in C++ PyTorch/XLA land.

upstream pr https://github.com/pytorch/pytorch/pull/112679#issuecomment-1789893320

### CODE REVIEW

**Review by wonjoolee95 — APPROVED** [review]
Thanks! Kokoro CI seems to fail because it can't recognize the new metric name, is this expected because it requires an upstream PyTorch pin?

**Comment by JackCaoG** [comment]
hmm, not sure I can rerun the CI once upstream thing merged

File: test/test_metrics.py [hunk 1]
```diff
@@ -38,6 +38,14 @@ def test_clear_metrics(self):
     self.assertIn("TensorToData", met.metrics_report())
     assert (len(met.metric_names()) > 0)
 
+  def test_tracing_time_metrics(self):
+    xla_device = xm.xla_device()
+    met.clear_all()
+    t1 = torch.tensor(156, device=xla_device)
+    t2 = t1 + 100
+    self.assertIn('LazyTracing', met.metric_names())
+    self.assertGreater(met.metric_data('LazyTracing')[0], 1)
```
Comment (alanwaketan) [inline test/test_metrics.py:47]: So the op counter will still be int? But adding another LazyTracing metrics?

Comment (JackCaoG) [reply test/test_metrics.py:47]: yea.

**Review by alanwaketan — APPROVED** [review]
LGTM.
